sound/soc/soc-component.c

```diff
@@ -297,39 +297,35 @@ EXPORT_SYMBOL_GPL(snd_soc_component_set_jack);
 int snd_soc_component_module_get(struct snd_soc_component *component,
 				 int upon_open)
 {
-	if (component->module)
+	if (component->module++)
```

Comment:
that ref count makes no sense to me (including in the original case). We should only increment in case there's no error, and in addition there are already module refcounts, not sure why we need new ones on top?

Reply:
@plbossart Ok let me remove. I tried first to not touch these, but then it completely broke driver load-reload... I'll try again.